### PR TITLE
Surface statistics

### PR DESCRIPTION
**Pull request by LorenzLamm**
This PR contains some example statistics that could be computed on a surface:
- protein density (#proteins / area)
- geodesic nearest neighbor distance
- angle between in-plane orientations of nearest neighbors

Not sure if these should go into Surforama or rather some downstream analysis package, but as discussed with @kevinyamauchi and @rdrighetto, it might be nice to have at least some basic statistics in there that could be automatically output.

I also added an example notebook that computes these statistics, based on a couple of membranes that I annotated (.obj and .star files; data available [here](https://drive.google.com/file/d/1iB2HRAKnM0slYKev0rr6c3DKvWrPCMpp/view?usp=sharing).

For geodesic distance computation, I added two alternative dependencies (I guess we should maybe decide for one of them?):
- [potpourri3d](https://pypi.org/project/potpourri3d/), which computes geodesic distance based on diffusion and is very fast, but apparently perfectly accurate
- [pygeodesic](https://pypi.org/project/pygeodesic/), which computes exact geodesic distances, but is much slower
In my experiments, the range of error was in the area [-0.2;0.2] voxels, and I would argue that the error in placing the points is higher. So my favorite would be potpourri3d, but happy to go with any of them (if we actually want to keep this functionality here).

Let me know what you think :)

PS: also added some minor fixes to Surforama:
- force orientations to be float (in my case, ROTATION values were cast to integers, leading to wrong orientations)
- hide slider values by default (only add them once "Start surfing" is activated

### CODE REVIEW

File: src/surforama/app.py [hunk 1]
```diff
@@ -437,7 +451,7 @@ def _find_point_on_click(self, layer, event):
         table_defaults[NAPARI_UP_0] = up_vector[0]
         table_defaults[NAPARI_UP_1] = up_vector[1]
         table_defaults[NAPARI_UP_2] = up_vector[2]
-        table_defaults[ROTATION] = 0
+        table_defaults[ROTATION] = 0.0
```
Comment (kevinyamauchi) [inline src/surforama/app.py:454]: Nice catch!

File: src/surforama/utils/stats.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+):
+    """
+    Compute the geodesic distance matrix between a set of points on a mesh.
+
+    Has two methods for computing the geodesic distances:
+    - "exact": Uses the exact geodesic algorithm from the pygeodesic library.
+        (takes forever for large meshes and many points)
+    - "fast": Uses the mesh heat method distance solver from the potpourri3d library.
+        (faster but less accurate -- but still good enough for most applications;
+        differences are in the area of ~[-0.2, 0.2] vx for the distances between points on the same mesh)
```
Comment (kevinyamauchi) [inline src/surforama/utils/stats.py:26]: I like having both options!

**Review by kevinyamauchi — APPROVED** [review]
Nice @LorenzLamm ! This looks good to me. I like that you started by adding the analysis code. We can always add GUI widgets, etc. in the future. By starting with the analysis functions, we can more quickly iterate. Thank you! I

In terms of the dependencies, let's just add both. I like having the option to have both the fast and exact versions.

In terms of whether the code belongs here or elsewhere, I think eventually, we would want to move the code to a more general purpose library. However, I think it makes sense to bundle it here for now until there's enough code that makes sense to split out.

in a follow up PR, it would be nice to add some plot widgets, etc. that are available from the GUI. Some sort of analysis widget or plugin. Lots of cool possibilities there!

**Comment by kevinyamauchi** [comment]
@LorenzLamm , since you're going to be away for a bit, I think we should just merge now and then continue iterating. What do you think? Are there any last features you want to add before you sign off?

**Comment by LorenzLamm** [comment]
> @LorenzLamm , since you're going to be away for a bit, I think we should just merge now and then continue iterating. What do you think? Are there any last features you want to add before you sign off?

Yes, I think we can merge.
There are some minor usability improvements that I would like to add after my annotation session and that I didn't have time to look into. But I guess this doesn't need to go into this PR anyways.
Will create an issue with this.

**Pull request merged**